pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -575,7 +575,7 @@ func (r *Reconciler) markAsDone(externalSecret *esv1beta1.ExternalSecret, start
 	SetExternalSecretCondition(externalSecret, *newReadyCondition)
 
 	externalSecret.Status.RefreshTime = metav1.NewTime(start)
-	externalSecret.Status.SyncedResourceVersion = util.GetResourceVersion(externalSecret.ObjectMeta)
+	externalSecret.Status.SyncedResourceVersion = util.GetResourceVersionExternalSecret(externalSecret)
```

Comment:
your implementation looks generally good! I think the only thing that we should try to strive for is using one single GetReosurceVersion. This could be acheived with some generics manipulation like this:

```
func GetResourceVersion[T v1beta1.ExternalSecretSpec | v1alpha1.PushSecretSpec](meta metav1.ObjectMeta, obj T) string {
	return fmt.Sprintf("%d-%s", meta.GetGeneration(), HashMeta(meta, obj))
}

func HashMeta(m metav1.ObjectMeta, obj any) string {
	type meta struct {
		annotations map[string]string
		labels      map[string]string
		spec        any
	}
	return utils.ObjectHash(meta{
		annotations: m.Annotations,
		labels:      m.Labels,
		spec:        obj,
	})
}
```

Reply:
Great, I overlooked this point before. let me update this point